### PR TITLE
[feature/bv-31] theme provider

### PR DESCRIPTION
**Pull request by dariuszsobkowicz**
- update material ui libs to latest
- theme object can be extracted in component by `useTheme` or `withTheme`
- default theme object (https://material-ui.com/customization/default-theme/) can be overridden to our needs
- constants are available in theme object

### CODE REVIEW

**Review by Milena-Wasiak — APPROVED** [review]
👍

**Review by wojciech-nowakowski — APPROVED** [review]

File: src/theme/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// default theme to override https://material-ui.com/customization/default-theme/
+
+import { createMuiTheme } from '@material-ui/core';
+import overrides from './overrides';
+import typography from './typography';
+import palette from './palette';
+import * as constans from './constants';
```
Comment (damianlewandowski) [inline src/theme/index.ts]: typo

File: src/theme/constants.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const DEFAULT_FONT_FAMILY = 'Quicksand';
+export const DEFAULT_FONT_SIZE = '0.875rem';
+export const DEFAULT_FONT_WIGHT = 'bold';
```
Comment (damianlewandowski) [inline src/theme/constants.ts]: typo

**Review by damianlewandowski — COMMENTED** [review]
What about changing `theme.palette.primary` to those dark colors in the sketches, and `theme.palette.secondary` to green ones so you could easily use those colors in components of your choosing instead of adding global styles to each specific components.
Think more like reusing components instead of reusing global styles.

File: src/theme/constants.ts (hunk 2 above)
Comment (dariuszsobkowicz) [reply src/theme/constants.ts]: @damianlewandowski well we can do this way but I think there will be cases where we will need to override

Comment (damianlewandowski) [reply src/theme/constants.ts]: I agree, I don't want to completely remove overrides, just give a way of using those dark-grey and green colors, which are main and secondary colors by design, freely.

For example when we have color "green" used both in button and in header. Let's not define two overrides for "button" AND for "header", let's instead have this color easily accessible from `theme.palette` and use it in JSS.

**Pull request merged**